### PR TITLE
chore: cherry-pick fixes for v1.1307.1 hotfix release

### PR DESCRIPTION
**Pull request by PeterSchafer**
## Summary

Cherry-picks all fixes listed in [CLI-1796](https://snyksec.atlassian.net/browse/CLI-1796) for the v1.1307.1 hotfix release, plus security fixes identified via commit-message keyword scan.

### Ticket PRs (8 PRs, 9 commits)
- #7177 — bump GAF to v0.18.2 for CLI-1732 scrub fix
- #7180 — report SNYK-CLI-0008 for repos with no testable projects [OSF-481]
- #7185 — upgrade bundled semver to 7.5.2 for CVE-2022-25883
- #7170 — automatic integration of language server
- #7191 — fix(errors): differentiate none errors better (2 commits)
- #7184 — restore moduleName, triageAdvice, functions_new in JSON [OSF-479]
- #7189 — handle .snyk policy edge cases [OSF-485]
- #7204 — bump cli-extension-axi

### Security fixes (3 commits)
- #7161 — bump cli-extension-iac-rules to fix vulnerabilities [IAC-3524]
- #7208 — bump gRPC dependency to fix CVE-2026-84304
- #7208 — bump js-yaml to fix CVE-2026-84375

### Additional PRs
- #7212 — feat: IANDT-27: capture contributors

### GAF release branch
GAF pointed at `release/1.1307` (`v0.18.3-0.20260904153233-0e7f03ec6034`), based on v0.18.2 with cherry-picked commits:
- contributor capture middleware (#703)
- regenerate testapi client OSF-479 (#714)
- IANDT-26 response parsing fix (#718)
- IANDT-27 post invoke hook (#721)

This excludes the TOON/HTML report rendering and other unrelated GAF main changes.

## Test plan
- [ ] CI passes on this branch
- [ ] Verify `snyk test` JSON output includes moduleName, triageAdvice, functions_new
- [ ] Verify .snyk policy edge cases handled correctly
- [ ] Verify no-testable-projects error code SNYK-CLI-0008
- [ ] Verify contributor capture is gated behind `INTERNAL_SNYK_CONTRIBUTORS_ENABLED`
- [ ] Build and smoke-test the binary

🤖 Generated with [Claude Code](https://claude.com/claude-code)

[CLI-1796]: https://snyksec.atlassian.net/browse/CLI-1796?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[OSF-481]: https://snyksec.atlassian.net/browse/OSF-481?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[OSF-479]: https://snyksec.atlassian.net/browse/OSF-479?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[OSF-485]: https://snyksec.atlassian.net/browse/OSF-485?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[IAC-3524]: https://snyksec.atlassian.net/browse/IAC-3524?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

---

> [!NOTE]
> **Medium Risk**
> Hotfix touches core exit-code/JSON error paths and bumps many framework extensions; regressions could affect CI scripts that depend on exit codes or JSON streams, though changes are mostly restoring prior behavior.
> 
> **Overview**
> **v1.1307.1 hotfix** cherry-picks regression fixes and security dependency updates, and refreshes release notes for this patch.
> 
> **`snyk test` behavior** is tightened in the Go v2 core: **no supported manifest files** again maps to catalog **`SNYK-CLI-0008`** with **exit code 3** (via `NewNoSupportedFilesFoundError` in the exit-code map). Post-run handling uses new **`IsFailure`** so exit **1** (findings) and **3** (unsupported projects) are not treated like hard failures—avoiding duplicate/corrupt JSON when a secondary network error occurs. **`snyk test --json`** error documents go to **stdout** instead of stderr in structured mode.
> 
> **Dependencies** bump **`go-application-framework`**, **`cli-extension-axi`**, **`cli-extension-os-flows`**, **`snyk-ls`**, and related transitive modules; Node side bumps **`snyk-docker-plugin`** to 9.20.0 and pins **`js-yaml`** overrides for CVE fixes. **`NETWORK_REQUEST_RETRY_ALLOWED_PATHS`** gets CLI defaults merged with user config.
> 
> **Docs**: adds **`snyk cos`** help (targets, scans, findings) and adjusts README links for sbom/aibom; **acceptance** resilience tests disable **`INTERNAL_SNYK_CONTRIBUTORS_ENABLED`** to avoid early network calls under slow-server simulation.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 57ba3e61190af67ce1b87a8a5c9a2971d953b94d. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: cliv2/go.mod [hunk 1]
```diff
@@ -15,38 +15,38 @@ require (
...
-	github.com/snyk/cli-extension-os-flows v0.0.0-20260821101812-a52aa2dd8c3b
+	github.com/snyk/cli-extension-iac-rules v0.0.0-20260818142329-df932cc1794f
+	github.com/snyk/cli-extension-os-flows v0.0.0-20260903073306-ad447d86c87d
 	github.com/snyk/cli-extension-sbom v0.0.0-20260818092356-16ee760085f9
 	github.com/snyk/cli-extension-secrets v0.0.0-20260811115637-ee872c26b0cd
 	github.com/snyk/code-client-go v1.31.8
 	github.com/snyk/container-cli v0.0.0-20260213211631-cd2b2cf8f3ea
 	github.com/snyk/error-catalog-golang-public v0.0.0-20260806122555-28dc45bbbde6
-	github.com/snyk/go-application-framework v0.16.1
+	github.com/snyk/go-application-framework v0.19.2-0.20260905103718-1db65578dda1
```
Comment (danskmt) [inline cliv2/go.mod]: @PeterSchafer I think this needs to be updated: it is using a specific commit reference

**Review by danskmt — COMMENTED** [review]
I added a comment + 2 observations:
1) it seems we have set `TEST_SNYK_IGNORE_LIST` - if the changes are touching one of the ignored tests, I would suggest unsetting this variable and running tests once again.

2) commits are marked as "unsigned" in this PR

**Review by MoisBrigitte — APPROVED** [review]

File: cliv2/go.mod (hunk 1 above)
Comment (PeterSchafer) [reply cliv2/go.mod]: this is a hotfix version of GAF that we need

File: binary-releases/RELEASE_NOTES.md [hunk 2]
```diff
@@ -1,26 +1,16 @@
...
-* **general**: Prevents a possible connection leak when the Snyk API returns an error response. ([8fd7d65](https://github.com/snyk/snyk/commit/8fd7d6544beab0fe609e823539e6373e058e7239))
+* **test**: `snyk test` on a repository with no supported manifest files again reports `SNYK-CLI-0008` and exits 3, instead of a generic `SNYK-CLI-0000` with exit 2. `snyk test --json` writes the error document to stdout as expected. ([2cbdbda](https://github.com/snyk/snyk/commit/2cbdbda2c572c6631aed05bde8d99a9359c1c63a))
+* **test**: Restores `moduleName`, `insights.triageAdvice`, and `functions_new` fields in `snyk test --json` output. ([8e250c6](https://github.com/snyk/snyk/commit/8e250c614707a67c39e80b01f2307e601d2e4991))
+* **test**: `.snyk` policy files are now handled correctly in the unified test flow -- empty, whitespace-only and comment-only policies, date-only timestamps, and other edge cases that previously caused incorrect results or failures. ([4215104](https://github.com/snyk/snyk/commit/4215104d3c9cf2d57089624d84cd39ce5cf6bbc3))
```
Comment (CatalinSnyk) [inline binary-releases/RELEASE_NOTES.md]: nit: Not sure if the users know what the `unified test flow` means. Maybe worth a tweak?

Comment (MoisBrigitte) [reply binary-releases/RELEASE_NOTES.md]: I agree, we could use the release notes from the PR:
`snyk test now reads .snyk policy files correctly.`

**Review by CatalinSnyk — APPROVED** [review]
